torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1429,6 +1429,22 @@ def _make_tensor_helper(shape, low=None, high=None):
 
     return samples
 
+
+def reference_sgn(x):
+    if x.dtype not in [np.complex64, np.complex128]:
+        return np.sign(x)
+
+    out = (x / np.abs(x))
```

Comment:
Would you elaborate on this? Why is `x / np.abs(x)` not sufficient?

Reply:
Doesn't handle the case when x is `0`. Will add a comment.
```python
>>> import numpy as np
>>> a = np.array(0)
>>> a/np.abs(a)
<stdin>:1: RuntimeWarning: invalid value encountered in true_divide
nan
```